ragnarok_bytes/src/from_bytes/implement.rs

```diff
@@ -65,16 +65,20 @@ impl<T: FromBytes, const SIZE: usize> FromBytes for [T; SIZE] {
 
 impl FromBytes for String {
     fn from_bytes<Meta>(byte_reader: &mut ByteReader<Meta>) -> ConversionResult<Self> {
-        let mut value = String::new();
+        let mut value = Vec::<u8>::new();
 
         while let Ok(byte) = byte_reader.byte::<Self>() {
             match byte {
                 0 => break,
-                byte => value.push(byte as char),
+                byte => value.push(byte),
             }
         }
+        let result = match String::from_utf8(value).map_err(|error| error.into_bytes()) {
```

Comment:
```suggestion
        let result = String::from_utf8(value)
            .map_err(|error| error.into_bytes())
            .unwrap_or_else(|error| error.iter().map(|byte| *byte as char).collect());
```

My linter proposes the following version.

Reply:
Wow, it seems syntactically better.

Reply:
Just as a heads up: You actually didn't pushed the change, I think.